### PR TITLE
Deprecate ORTModel class

### PR DESCRIPTION
**Pull request by IlyasMoutawwakil**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/onnxruntime/utils.py [hunk 1]
```diff
@@ -341,3 +351,53 @@ class ORTQuantizableOperator(Enum):
     Resize = "Resize"
     AveragePool = "AveragePool"
     Concat = "Concat"
+
+
+def evaluation_loop(
```
Comment (echarlaix) [inline optimum/onnxruntime/utils.py:356]: should we also update the examples to directly use `evaluation_loop`

https://github.com/huggingface/optimum/blob/f755a58e56597f690be4a0c4bdb549ce0ffd4e03/examples/onnxruntime/optimization/question-answering/run_qa.py#L487

**Review by echarlaix — APPROVED** [review]
Thanks a lot @IlyasMoutawwakil !!

File: optimum/onnxruntime/model.py [hunk 2]
```diff
@@ -49,6 +49,11 @@ def __init__(
             label_names (`List[str]`, `optional`):
                 The list of keys in your dictionary of inputs that correspond to the labels.
         """
+
+        logger.warning(
+            "The class `optimum.onnxruntime.model.ORTModel` is deprecated and will be removed in the next release."
```
Comment (echarlaix) [inline optimum/onnxruntime/model.py:54]: It shouldn't be used anywhere so no strong opinion on this, but we can remove it in v2.0 just to be safe @fxmarty wdyt ?

Comment (fxmarty) [reply optimum/onnxruntime/model.py:54]: No strong opinion on this either, it sounds fine to me to merge it now as well.

**Pull request merged**